### PR TITLE
Feat/4820 button inline link

### LINKED ISSUE #4820: Inline link variation of button component that acts like a link
https://github.com/patternfly/patternfly/pull/3470/

This is a variation of an inline link button that behaves like regular link text. We can't use a native `<button>` html element and have text display inline and wrap to multiple lines like regular text/links, so if a user has an inline link button that is long and they want to wrap like regular text, we need to use a `<span>` instead of `<button>` with `role="button"` and `tabindex="0"` so it's focusable. When it's disabled, we need to use `pf-m-disabled` and set `aria-disabled="true"` and `tabindex="-1"`, similar to when an `<a>` is used as the lement for the button component.

### PR DESCRIPTION
**Pull request by evwilkin**
**What**: Closes #4820 

This PR adds the inline link as span, which uses a `span` with `role="button"` to allow text to wrap as a normal inline link would.

### CODE REVIEW

File: packages/react-core/src/components/Button/examples/Button.md [hunk 1]
```diff
@@ -168,6 +168,22 @@ LinkButton = () => (
 );
 ```
 
+## Inline link as span
```
Comment (redallen) [inline packages/react-core/src/components/Button/examples/Button.md:171]: This should be an h3. Change to `### Inline link as span` to fix the "Untitled example":
image

File: packages/react-core/src/components/Button/examples/Button.md [hunk 2]
```diff
@@ -168,6 +168,22 @@ LinkButton = () => (
...
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+
+InlineLinkAsSpan = () => (
+  <p>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+    <Button variant="link" isInline component="span">
+      This is long button text that needs to be a span so that it will wrap inline with the text around it.
+    </Button>
+    Sed hendrerit nisi in cursus maximus. Ut malesuada nisi turpis, in condimentum velit elementum non.
```
Comment (redallen) [inline packages/react-core/src/components/Button/examples/Button.md:182]: Change to `{' '}Sed ...` to preserve whitespace.

**Review by redallen — CHANGES_REQUESTED** [review]
Implementation looks good, docs just need fixing.

File: packages/react-core/src/components/Button/examples/Button.md (hunk 1 above)
Comment (evwilkin) [reply packages/react-core/src/components/Button/examples/Button.md]: Updated - thanks!

File: packages/react-core/src/components/Button/examples/Button.md (hunk 2 above)
Comment (evwilkin) [reply packages/react-core/src/components/Button/examples/Button.md]: Updated

File: yarn.lock [hunk 3]
```diff
@@ -209,7 +209,7 @@
     "@babel/helper-explode-assignable-expression" "^7.10.4"
     "@babel/types" "^7.10.4"
 
-"@babel/helper-builder-react-jsx-experimental@^7.10.4", "@babel/helper-builder-react-jsx-experimental@^7.11.5":
```
Comment (redallen) [inline yarn.lock:212]: Can you remove this file by running `git checkout upstream/master -- yarn.lock`?

**Review by redallen — CHANGES_REQUESTED** [review]

**Review by redallen — DISMISSED** [review]
LGTM!

File: yarn.lock (hunk 3 above)
Comment (evwilkin) [reply yarn.lock]: Updated - thanks for the help

**Review by jessiehuff — APPROVED** [review]
LGTM! 😄 Only thing to note here is that because it's not an actual button, the onclick event will not fire on enter and spacebar keys with a `<span>`. That means you have to add an onkeydown event and check if the enter or spacebar keys are pressed, then prevent the default behavior for the spacebar key and run the code.
Here is a [resource](https://www.deque.com/blog/accessible-aria-buttons/#:~:text=Keyboard%20Focusability,-HTML%20Native%20Controls&text=You%20need%20a%20for,to%20make%20it%20keyboard%20operable.) where Deque goes over these considerations. The section I'm referring to is "Custom ARIA `<span role=”button”>`."

**Comment by evwilkin** [comment]
@tlabaj wanted to get your feedback here - is this something we should note for users, perhaps by adding a quick blurb beneath the new `Inline link as span` example?

> LGTM! 😄 Only thing to note here is that because it's not an actual button, the onclick event will not fire on enter and spacebar keys with a `<span>`. That means you have to add an onkeydown event and check if the enter or spacebar keys are pressed, then prevent the default behavior for the spacebar key and run the code.

**Comment by evwilkin** [comment]
@jessiehuff @tlabaj thanks for the feedback - I've added an example & text within it explaining how to use the `onKeyDown` listener when the using the inline link as span to maintain keyboard interactions.

**Review by jessiehuff — DISMISSED** [review]
LGTM! Thanks! 😄

**Review by mcoker — DISMISSED** [review]
Nice!! LGTM!

**Review by redallen — DISMISSED** [review]

**Review by redallen — APPROVED** [review]

**Review by jessiehuff — APPROVED** [review]

**Pull request merged**